### DISCUSSION TITLE
Change embeddings

### DISCUSSION THREAD

**Discussion by aqred1** [facebookresearch/esm#65] [Q&A] [answered]
Hi all,

Congratulations on this fantastic innitiative.
This is a general question. I was wondering if we could use the pre-trained models but using different embeddings, without retraining the model, perhaps only finetuning. So a way to disentangle the Transformer layers from the embedding layer. I guess not, right?

Thanks.

**Comment by tomsercu** [comment, marked as answer]
You could do it in principle, but doesn't seem very well motivated at first sight since you'd change the input without giving the model a chance to adapt. Maybe more straightforward is to keep the trunk of the model frozen (initial `nn.Embedding`, first X layers), and finetune only the last few (33-X for ESM-1b) layers.

**Reply by aqred1** [reply]
Thanks, that makes sense.
